parts/linux/cloud-init/artifacts/ensure-no-dup.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# remove this if we are no longer using promiscuous bridge mode for containerd
+# background: we get duplicated packets from pod to serviceIP if both are on the same node (one from the cbr0 bridge and one from the pod ip itself via kernel due to promiscuous mode being on)
+# we should filter out the one from pod ip
+# this is exactly what kubelet does for dockershim+kubenet
+# https://github.com/kubernetes/kubernetes/pull/28717
+
+ebtables -t filter -L AKS-DEDUP 2>/dev/null
+if [[ $? -eq 0 ]]; then
+    echo "AKS-DEDUP rule already set"
+    exit 0
+fi
+if [[ ! -f /etc/cni/net.d/10-containerd-net.conflist ]]; then
+    echo "cni config not up yet...exiting early"
+    exit 1
+fi
+podSubnetAddr=$(cat /etc/cni/net.d/10-containerd-net.conflist  | jq -r ".plugins[] | select(.type == \"bridge\") | .ipam.subnet")
+
+if [[ ! -f /sys/class/net/cbr0/address ]]; then
+    echo "cbr0 bridge not up yet...exiting early"
+    exit 1
+fi
+cbr0MAC=$(cat /sys/class/net/cbr0/address)
+
+cbr0IP=$(ip addr show cbr0 | grep -Eo "inet ([0-9]*\.){3}[0-9]*" | grep -Eo "([0-9]*\.){3}[0-9]*")
+if [[ -z "${cbr0IP}" ]]; then
+    echo "cbr0 bridge does not have an ipv4 address...exiting early"
+    exit 1
+fi
+
+echo "adding AKS-DEDUP ebtable chain"
+ebtables -t filter -N AKS-DEDUP # add new AKS-DEDUP chain
+
+echo "adding rule: ebtables -t filter -A AKS-DEDUP -p IPv4 -s ${cbr0MAC} -o veth+ --ip-src ${cbr0IP} -j ACCEPT"
+ebtables -t filter -A AKS-DEDUP -p IPv4 -s ${cbr0MAC} -o veth+ --ip-src ${cbr0IP} -j ACCEPT
+
+echo "adding rule: ebtables -t filter -A AKS-DEDUP -p IPv4 -s ${cbr0MAC} -o veth+ --ip-src ${podSubnetAddr} -j DROP"
```

Comment:
instead of echoing here, I'm thinking about printing all AKS_DEDUP rules after applying them, that will give us a firm confirmation on the success.